src/BloomExe/web/ServerBase.cs

```diff
@@ -116,117 +74,122 @@ private bool StartWithExceptionHandling(out Exception error)
 			catch (Exception e)
 			{
 				error = e;
-				_listener = null;
 				return false;
 			}
 		}
 
 		private bool TryStart()
 		{
-			_listener = new HttpListener();
 			_listener.AuthenticationSchemes = AuthenticationSchemes.Anonymous;
 			_listener.Prefixes.Add(PathEndingInSlash);
 			_listener.Start();
-			//			_listener.BeginGetContext(new AsyncCallback(ListenerCallback), _listener);
 
-			_listenerThread = new Thread(HandleRequests);
 			_listenerThread.Start();
 
+			for (var i = 0; i < _workers.Length; i++)
+			{
+				_workers[i] = new Thread(WaitForRequests);
+				_workers[i].Start();
+			}
+
 			return GetIsAbleToUsePort();
 		}
 
-		private bool GetIsAbleToUsePort()
+		private void HandleRequests()
```

Comment:
I'm slowly getting it, but this class badly needs a comment that outlines how the pieces work together. It would help to have a comment on this method that says something like "This method is what the main listenerThread does: sit in a loop in which it gets an HTTP request from the _listener and queues it for handling by one of our workers"